### PR TITLE
adding initial vsphere tf files

### PR DESCRIPTION
**Pull request by dav1x**
The following will provision infrastructure based on provided variables.tf:

1. vCenter resource pool
2. X number of VMs with IP assignments (only with vmware guest customizations)
3. This IP customization will need to be modified for RHOCS

### CODE REVIEW

**Comment by staebler** [comment]
@dav1x This cannot be merged until it passes `/test tf-fmt`.

**Comment by dav1x** [comment]
@staebler, sorry about that. I'll fix that today.

File: upi/vsphere/bootstrap/main.tf [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  name                    = "${var.resource_pool}"
+  parent_resource_pool_id = "${data.vsphere_compute_cluster.compute_cluster.resource_pool_id}"
+}
+
+resource "vsphere_virtual_machine" "vm" {
+  name             = "bootstrap-0"
+  resource_pool_id = "${vsphere_resource_pool.resource_pool.id}"
+  datastore_id     = "${data.vsphere_datastore.datastore.id}"
+  num_cpus         = 4
+  memory           = 8192
```
Comment (rphillips) [inline upi/vsphere/bootstrap/main.tf]: bootstrap machine can mostly likely be 2048 memory and 2 cpus (same as libvirt).

File: upi/vsphere/masters/main.tf [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  parent_resource_pool_id = "${data.vsphere_compute_cluster.compute_cluster.resource_pool_id}"
+}
+
+resource "vsphere_virtual_machine" "vm" {
+  count            = 4
+  name             = "ocp${count.index + 1}"
+  resource_pool_id = "${vsphere_resource_pool.resource_pool.id}"
+  datastore_id     = "${data.vsphere_datastore.datastore.id}"
+  num_cpus         = 4
+  memory           = 8192
```
Comment (rphillips) [inline upi/vsphere/masters/main.tf]: making the master cpus and memory configurable with a variable would be nice.

File: upi/vsphere/bootstrap/main.tf (hunk 1 above)
Comment (dav1x) [reply upi/vsphere/bootstrap/main.tf]: Agreed! Parameterized CPU and memory. I haven't started writing that section yet. But, I'll pass the appropriate sizing.

File: upi/vsphere/masters/main.tf (hunk 2 above)
Comment (dav1x) [reply upi/vsphere/masters/main.tf]: Agreed! Parameterized CPU and memory and set these as defaults in variables.tf

**Pull request merged**